### PR TITLE
Lesson6

### PR DESCRIPTION
**Pull request by gxst7**
add tasks for lesson 6
upgrade tasks at lesson 5

### CODE REVIEW

File: src/com/tms/lesson5/Director.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package com.tms.lesson5;
+
+import com.tms.lesson6.Searchable;
+
+import java.util.ArrayList;
+
+public class Director extends Employee implements Searchable {
+
+    private ArrayList<Employee> employeesForTheDirector = new ArrayList<>();
```
Comment (gr00shik) [inline src/com/tms/lesson5/Director.java]: Left - abstract, right - implementation
List = ArrayList

File: src/com/tms/lesson6/Searchable.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.tms.lesson6;
+
+import com.tms.lesson5.Director;
+import com.tms.lesson5.Employee;
+
+public interface Searchable {
```
Comment (gr00shik) [inline src/com/tms/lesson6/Searchable.java]: no need to use both methods: getEmployeeByName  and getEmployeeByNameDeep - only one.
Implementation will use different realisations.
SimpleSearch implements Searchable
DeepSearch implements Searchable

File: src/com/tms/lesson5/Director.java (hunk 1 above)
Comment (gxst7) [reply src/com/tms/lesson5/Director.java]: fixed

File: src/com/tms/lesson6/Searchable.java (hunk 2 above)
Comment (gxst7) [reply src/com/tms/lesson6/Searchable.java]: fixed

**Pull request merged**